src/univariate/continuous/laplace.jl

```diff
@@ -136,3 +136,15 @@ function fit_mle(::Type{<:Laplace}, x::AbstractArray{<:Real})
     xc .= abs.(x .- m)
     return Laplace(m, mean(xc))
 end
+
+function fit_mle(::Type{<:Laplace}, x::AbstractArray{<:Real}, w::AbstractArray{<:Real})
+    n = length(x)
+    if n != length(w)
+        throw(DimensionMismatch("Inconsistent array lengths."))
+    end
+    xc = similar(x)
+    copyto!(xc, x)
+    m = median(xc, weights(w)) # https://github.com/JuliaStats/StatsBase.jl/blob/0b64a9c8a16355da16469d0fe5016e0fdf099af5/src/weights.jl#L788
+    xc .= abs.(x .- m)
+    return Laplace(m, mean(xc, weights(w)))
+end
```

Comment:
Actually, I've taken the opportunity to remove an unneeded allocation; there's no need to create an extra array for the centered variables.
```suggestion
function fit_mle(::Type{<:Laplace}, data::AbstractArray{<:Real})
    med = median(data)
    # TODO: is it possible to do a single-pass median and mean absolute deviation?
    mean_abs_deviation = mean(x) do x
        return abs(x - mean_abs_deviation)
    end
    return Laplace(med, mean_abs_dev)
end

function fit_mle(::Type{<:Laplace}, x::AbstractArray{<:Real}, w::AbstractWeights)
    if length(x) != length(w)
        throw(DimensionMismatch("Inconsistent array lengths."))
    end
    # TODO: can this be done in a single-pass?
    # code for weighted median can be found at:
    # https://github.com/JuliaStats/StatsBase.jl/blob/0b64a9c8a16355da16469d0fe5016e0fdf099af5/src/weights.jl#L788
    med = median(xc, weights(w))
    norm = sum(w)
    mean_abs_deviation = mean(zip(x, weights)) / norm do x, weight
        return weight * abs(x - med)
    end
    return Laplace(med, mean_abs_deviation)
end
```

Reply:
The suggestion breaks the non-weighted implementation without improving its allocations (I don't think the non-weighted algorithm can be improved allocationwise).

Reply:
> The suggestion breaks the non-weighted implementation without improving its allocations

I made a mistake earlier. Not sure if it works now, but @dmetivie can double-check hopefully. (Is there an easy way to run tests on suggestions?)


> I don't think the non-weighted algorithm can be improved allocationwise

The code for both algorithms is the same. But in any case, there's definitely no need to copy an array just to center it.

Reply:
> But in any case, there's definitely no need to copy an array just to center it.

The code on master seems to be as efficient as possible: The copy is used by the median calculations and then only reused by the centering operation. Note that the code uses `median!` - if you switch to `median`, a copy is created internally and discarded afterwards. So IMO the current code should not be changed.